### PR TITLE
gitbase: skip hash references that do not point to commits

### LINKED ISSUE #213: Squashed Join: Object not found error
`Object not found error` happens sometimes using the squash joins rule. It appears to be because references can point to any object type, not only commits. Check if we are ignoring `object not found` error when a reference is pointing to a not-a-commit object.

### PR DESCRIPTION
**Pull request by erizocosmico**
Fixes #213

### CODE REVIEW

File: iterator.go [hunk 1]
```diff
@@ -791,6 +791,10 @@ func (i *refCommitsIter) Advance() error {
 				return io.EOF
 			}
 
+			if err == plumbing.ErrObjectNotFound {
```
Comment (kuba--) [inline iterator.go]: lets reverse order of all these checks (why do we check EOF and Object not found if an error is nil?
```go
if err != nil {
if err == io.EOF {
 				i.refs = nil
  				return io.EOF
  			}
if err == plumbing.ErrObjectNotFound {
  continue
}
return err
```

**Comment by ajnavarro** [comment]
I think we should do something like here: https://github.com/src-d/borges/blob/master/git.go#L219

Object not found can be for other reasons too.

**Comment by erizocosmico** [comment]
@ajnavarro Sounds good. Should we resolve the tag to a commit like this does, too, instead of skipping it?

**Comment by erizocosmico** [comment]
Fixed (waiting on Antonio to tell if we should resolve the tags to a commit before adding that)

**Comment by ajnavarro** [comment]
@erizocosmico yes, we should resolve tags too to commits if possible

**Pull request merged**